src/rlic/_boundaries.py

```diff
@@ -0,0 +1,39 @@
+from typing import NewType, TypeAlias, TypedDict
+
+Bound = NewType("Bound", str)
+BoundPair = tuple[Bound, Bound]
+AnyBound: TypeAlias = Bound | BoundPair
+
+
+class AbstractBounds(TypedDict):
+    udir: AnyBound
+    vdir: AnyBound
+
+
+class ConcreteBounds(TypedDict):
+    udir: BoundPair
+    vdir: BoundPair
+
+
+def as_pair(b: AnyBound) -> BoundPair:
+    if isinstance(b, tuple):
```

Comment:
I should probably use pattern matching everywhere

Reply:
... but I also should be very strict on the return type being a tuple (not just something that matches the structure, e.g. a list)